### PR TITLE
feat: add app monitor for wso2

### PR DESCRIPTION
**Pull request by GraceRuan**

### CODE REVIEW

File: terraform/app-monitor-module/variables.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+variable "app_monitor" {
+  type = object({
+    name =string
+    server = string
+    app = string
+    query_level_trigger_id = string
+    automation_queue_action_id = string
+    errormsg = string
```
Comment (mbystedt) [inline terraform/app-monitor-module/variables.tf:8]: Maybe rename this (and in the json file) as query?

File: terraform/main.tf [hunk 2]
```diff
@@ -566,6 +566,14 @@ module "agent-monitor" {
   automation_destination_id = module.destination["message-queue"].destination_id
 }
 
+module "app-monitor" {
+  source = "./app-monitor-module"
+  for_each = { for a in jsondecode(file("./app-alert.json")): a.name => a }
+  app_monitor = each.value
+  depends_on = [aws_opensearch_domain.es]
+  automation_destination_id = module.destination["wso2-alert"].destination_id
```
Comment (mbystedt) [inline terraform/main.tf:574]: What if you don't want to use this destination?

JSON probably needs to include a bunch more details like the destination (interval, throttle, action(s), templates)

**Review by mbystedt — CHANGES_REQUESTED** [review]

File: terraform/main.tf (hunk 2 above)
Comment (GraceRuan) [reply terraform/main.tf]: Use yaml as recommended and use queue name for destination.

File: terraform/app-monitor-module/variables.tf (hunk 1 above)
Comment (GraceRuan) [reply terraform/app-monitor-module/variables.tf]: set queryblock for query part. Addition to that, use more fields for monitor config

**Review by mbystedt — APPROVED** [review]
Nice work

**Pull request merged**